fix: Properly parse multi-value shorthand with slash

yasuhiro-yamamoto opened the pull request:
## What changed / motivation ?

Fixed the expansion handling of border-radius shorthand values containing the '/' separator.

## Linked PR/Issues

Fixes #730 

## Additional Context

- Modified the splitValue function to handle nodes containing '/' by merging them with adjacent nodes
- Example: '1px / 2px' is now correctly processed as borderStartStartRadius: '1px / 2px'
- Added test cases in packages/shared/__tests__/split-value-test.js

## Pre-flight checklist

- [x] I have read the contributing guidelines
      [Contribution Guidelines](https://github.com/facebook/stylex/blob/main/.github/CONTRIBUTING.md)
- [x] Performed a self-review of my code

nmn commented on packages/shared/__tests__/split-value-test.js:
```diff
@@ -44,4 +44,13 @@ describe('Ensure CSS values are split correctly', () => {
       splitValue('calc((100% - 50px) * 0.5) var(--rightpadding, 20px)'),
     ).toEqual(['calc((100% - 50px) * 0.5)', 'var(--rightpadding,20px)']);
   });
+
+  test('Splits a string of values with slash notation appropriately.', () => {
+    expect(splitValue('1px/2px 3px 4px 5px')).toEqual([
+      '1px/2px',
+      '3px',
+      '4px',
+      '5px',
+    ]);
+  });
 });
```
This is not how border-radius values are expanded. The correct expansion should be:


```suggestion

  test('Splits a string of values with slash notation appropriately.', () => {
    expect(splitValue('1px/2px 3px 4px 5px')).toEqual([
      '1px 2px',
      '1px 3px',
      '1px 4px',
      '1px 5px',
    ]);
  });
});
```

See https://developer.mozilla.org/en-US/docs/Web/CSS/border-radius
which mentions:

```css
border-radius: 4px 3px 6px / 2px 4px;

/* It is equivalent to: */
border-top-left-radius: 4px 2px;
border-top-right-radius: 3px 4px;
border-bottom-right-radius: 6px 2px;
border-bottom-left-radius: 3px 4px;
```

yasuhiro-yamamoto replied:
I misunderstood the usage of `/` in `border-radius`. I will review the specification and correct the code accordingly.

nmn reviewed (changes requested):
Thanks for trying to fix this edge case, but the logic isn't quite right.

(Note: also, this code path is only used if you use the `legacy-expand-shorthands` configuration which should not be needed in the vast majority of use-cases.)

yasuhiro-yamamoto commented:
@nmn I've revised the expansion logic for cases involving '/' and corrected the code. Additionally, I've updated the `splitValue` function to accept an optional argument, and explicitly handle processing specific to border-radius within the code.

nmn reviewed (approved):
LGTM.

nmn commented:
Will merge this PR after doing more testing as this code primarily affects internal code.
